### PR TITLE
Upgrading to use erlang 23

### PR DESCRIPTION
**Pull request by alanz**
### Description

Upgrade to use Erlang / OTP 23

### CODE REVIEW

**Comment by alanz** [comment]
Currently dialyzer checks fail for getting the EEP48 doc stuff, struggling with
it.

File: src/els_uri.erl [hunk 1]
```diff
@@ -69,7 +69,7 @@ uri(Path) ->
   els_utils:to_binary(
     uri_string:recompose(#{
       scheme => <<"file">>,
-      host => http_uri:encode(Host),
+      host => Host,
```
Comment (alanz) [inline src/els_uri.erl:72]: My reading of [uri_string:recompose/1](https://erlang.org/doc/man/uri_string.html#recompose-1) is that *it* will do all necessary encoding, we do not need to do it for the sub-components.

File: src/els_hover_provider.erl [hunk 2]
```diff
@@ -74,23 +74,30 @@ documentation(_M, _POI) ->
...
+          FuncDoc ->
+            #{ kind => content_kind()
+             , value => els_utils:to_binary(FuncDoc)
+             }
+        end;
+      _R1 ->
+        docs_from_src(M, F, A)
+    end
+  catch C:E ->
+      lager:error("[hover] Error fetching EEP48 docs [error=~p]", [{C, E}]),
```
Comment (alanz) [inline src/els_hover_provider.erl:100]: This generates the log line 

```
2020-05-20 11:29:20.398 [error] <0.184.0>@els_hover_provider:get_docs:99 [hover] Error fetching EEP48 docs [error={error,{case_clause,{error,beam_lib,{file_error,"/usr/local/bin/erlang_ls/erlang_ls/ebin/els_hover_provider.beam",enotdir}}}}]
```

**Comment by alanz** [comment]
Current dialyzer results

```
rebar3 dialyzer
Verifying dependencies...
Compiling erlang_ls
Dialyzer starting, this may take a while...
Updating plt...
Resolving files...
Checking 753 files in "/Users/alanzimm/repos/erlang_ls/erlang_ls/_build/default/rebar3_23.0_plt"...
Doing success typing analysis...
Resolving files...
Analyzing 72 files with "/Users/alanzimm/repos/erlang_ls/erlang_ls/_build/default/rebar3_23.0_plt"...

src/els_hover_provider.erl
  79: Invalid type specification for function els_hover_provider:get_docs/3. The success typing is (atom(),atom(),byte()) -> <<>>
  88: The call shell_docs:render(M::atom(), F::atom(), A::byte(), DocChunk::#docs_v1{format::<<_:184>>}) breaks the contract (Module::module(), Function::fun(), Arity::arity(), Docs::docs_v1()) -> unicode:chardata() | {'error','function_missing'}
Warnings written to /Users/alanzimm/repos/erlang_ls/erlang_ls/_build/default/23.0.dialyzer_warnings
Warnings occurred running dialyzer: 2

```

File: src/els_uri.erl (hunk 1 above)
Comment (robertoaloi) [reply src/els_uri.erl:72]: In my branch (I was doing the same), I just stripped it away.

File: src/els_hover_provider.erl (hunk 2 above)
Comment (robertoaloi) [reply src/els_hover_provider.erl:100]: This is the bug I was talking about. I am already addressing this in a local branch. Will push it today.

Comment (robertoaloi) [reply src/els_hover_provider.erl:100]: And the related "echo, ok" error you will eventually encounter :)

Comment (alanz) [reply src/els_hover_provider.erl:100]: well, I will leave it to you then, good luck keeping dialyzer happy.  Will revert to what I was *actually* trying to do.

Comment (robertoaloi) [reply src/els_hover_provider.erl:100]: Didn't mean to scare you away :) It's the group leader fix I mentioned at the weekly sync.

Comment (alanz) [reply src/els_hover_provider.erl:100]: Well, I was not scared away, per se, but just figured if you already have a branch, further along than mine, there is no point in us duplicating work.

Especially as my current interest is more toward the older versions anyway.

**Comment by robertoaloi** [comment]
Closing as superseded by #551

**Pull request closed without merging**